webui/react/src/components/UPlot/UPlotChart.tsx

```diff
@@ -63,145 +67,98 @@ const shouldUpdate = (
-    const trimmedData = chartData.map((series) => series.slice(0, minDataLength));
-
-    // Checking to make sure the X series has some data.
-    const hasXValues = (trimmedData?.[0]?.length ?? 0) !== 0;
-
-    return [ hasXValues, trimmedData as unknown as AlignedData ];
-  }
+const shouldRecreate = (
+  chart?: uPlot,
+  options?: Partial<uPlot.Options>,
+  prevOptions?: Partial<uPlot.Options>,
+): boolean => {
+  return true;
 };
-const SCROLL_THROTTLE_TIME = 500;
 
 const UPlotChart: React.FC<Props> = ({ data, focusIndex, options, style }: Props) => {
   const chartRef = useRef<uPlot>();
-  const [ chartIsMounted, setChartIsMounted ] = useState(false);
   const chartDivRef = useRef<HTMLDivElement>(null);
   const scalesZoomData = useRef<Record<string, ScaleZoomData>>({});
-
-  const getAugmentedOptions = (options: Partial<uPlot.Options> | undefined)
-  : uPlot.Options => uPlot.assign(
-    {
-      hooks: {
-        setScale: [ (uPlot: uPlot, scaleKey: string) => {
-          if (![ 'x', 'y' ].includes(scaleKey)) return;
-
-          const currentMax: number|undefined =
-            uPlot.posToVal(scaleKey === 'x' ? uPlot.bbox.width : 0, scaleKey);
-          const currentMin: number|undefined =
-            uPlot.posToVal(scaleKey === 'x' ? 0 : uPlot.bbox.height, scaleKey);
-          let max: number|undefined = scalesZoomData.current[scaleKey]?.max;
-          let min: number|undefined = scalesZoomData.current[scaleKey]?.min;
-
-          if (max == null || currentMax > max) max = currentMax;
-          if (min == null || currentMin < min) min = currentMin;
-
-          scalesZoomData.current[scaleKey] = {
-            isZoomed: currentMax < max || currentMin > min,
-            max,
-            min,
-          };
-        } ],
-      },
-      width: chartDivRef.current?.offsetWidth,
+  const dataRef = useRef<uPlot.AlignedData>(EMPTY_DATA);
+  const [ isEmpty, setIsEmpty ] = useState(true);
+
+  const fullOptions = useMemo(() => uPlot.assign({
+    hooks: {
+      setScale: [ (uPlot: uPlot, scaleKey: string) => {
+        if (![ 'x', 'y' ].includes(scaleKey)) return;
+
+        const boundWidth = scaleKey === 'x' ? uPlot.bbox.width : 0;
+        const boundHeight = scaleKey === 'x' ? 0 : uPlot.bbox.height;
+        const currentMax: number | undefined = uPlot.posToVal(boundWidth, scaleKey);
+        const currentMin: number | undefined = uPlot.posToVal(boundHeight, scaleKey);
+        let max: number | undefined = scalesZoomData.current[scaleKey]?.max;
+        let min: number | undefined = scalesZoomData.current[scaleKey]?.min;
+
+        if (max == null || currentMax > max) max = currentMax;
+        if (min == null || currentMin < min) min = currentMin;
+
+        scalesZoomData.current[scaleKey] = {
+          isZoomed: currentMax < max || currentMin > min,
+          max,
+          min,
+        };
+      } ],
     },
-    options || {},
-  ) as uPlot.Options;
+  }, options ?? {}) as uPlot.Options, [ options ]);
```

Comment:
`options` is constantly getting reassigned, so I don't think the `useMemo` is doing anything for us here

Reply:
and even it was doing something we could be closing over stale values from `scalesZoomData`

Reply:
^^wait is that true though? My thinking was that since max/min are numbers they will get passed in by value and not change if the ref changes, but I am not 100%

Also RE: options getting reassigned: it looks like most places besides the profiler that is not happening. I am working on a refactor of the profiler metrics themselves that allows us to avoid reassigning options, so this would be useful if we can land that.